src/modules/git_status.rs

```diff
@@ -0,0 +1,122 @@
+use ansi_term::Color;
+use git2::{Repository, Status};
+
+use super::{Context, Module};
+
+/// Creates a segment with the Git branch in the current directory
+///
+/// Will display the branch name if the current directory is a git repo
+pub fn segment(context: &Context) -> Option<Module> {
+    // This is the order that the sections will appear in
+    const GIT_STATUS_CONFLICTED: &str = "=";
+    const GIT_STATUS_AHEAD: &str = "⇡";
+    const GIT_STATUS_BEHIND: &str = "⇣";
+    const GIT_STATUS_DIVERGED: &str = "⇕";
+    const GIT_STATUS_UNTRACKED: &str = "?";
+    const GIT_STATUS_STASHED: &str = "$";
+    const GIT_STATUS_MODIFIED: &str = "!";
+    const GIT_STATUS_ADDED: &str = "+";
+    const GIT_STATUS_RENAMED: &str = "»";
+    const GIT_STATUS_DELETED: &str = "✘";
+
+    let branch_name = context.branch_name.as_ref()?;
+    let repo_root = context.repo_root.as_ref()?;
+    let repository = Repository::open(repo_root).ok()?;
+
+    let module_style = Color::Red.bold();
+    let mut module = Module::new("git_status");
+    module.get_prefix().set_value("[").set_style(module_style);
+    module.get_suffix().set_value("] ").set_style(module_style);
+    module.set_style(module_style);
+
+    let ahead_behind = get_ahead_behind(&repository, &branch_name);
+    log::debug!("Repo ahead/behind: {:?}", ahead_behind);
+    let stash_object = repository.revparse_single("refs/stash");
+    log::debug!("Stash object: {:?}", stash_object);
+    let repo_status = get_repo_status(&repository);
+    log::debug!("Repo status: {:?}", repo_status);
```

Comment:
This is cool will make debugging a lot easier 👍 Just want to double check if its still in there on purpose :)

Reply:
Yep, it's in there on purpose. 😄
I'd like to eventually go back and add detailed trace and debug logs to make it clear what the prompt is doing under the hood.